packages/markers/src/browser/problem/problem-composite-tree-node.spec.ts

```diff
@@ -0,0 +1,310 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { enableJSDOM } from '@theia/core/lib/browser/test/jsdom';
+let disableJSDOM = enableJSDOM();
+
+import URI from '@theia/core/lib/common/uri';
+import { expect } from 'chai';
+import { Container } from '@theia/core/shared/inversify';
+import { Diagnostic, Range, DiagnosticSeverity } from '@theia/core/shared/vscode-languageserver-types';
+import { Event } from '@theia/core/lib/common/event';
+import { MarkerManager } from '../marker-manager';
+import { MarkerInfoNode, MarkerNode, MarkerOptions, MarkerRootNode } from '../marker-tree';
+import { PROBLEM_OPTIONS } from './problem-container';
+import { ProblemManager } from './problem-manager';
+import { ProblemTree } from './problem-tree-model';
+import { FileService } from '@theia/filesystem/lib/browser/file-service';
+import { ProblemCompositeTreeNode } from './problem-composite-tree-node';
+import { Marker } from '../../common/marker';
+
+disableJSDOM();
+
+let rootNode: MarkerRootNode = {
+    visible: false,
+    id: 'theia-' + 'problem' + '-marker-widget',
+    name: 'MarkerTree',
+    kind: 'problem',
+    children: [],
+    parent: undefined
+};
+
+before(() => {
+    disableJSDOM = enableJSDOM();
+    const testContainer = new Container();
+
+    testContainer.bind(MarkerManager).toSelf().inSingletonScope();
+    testContainer.bind(ProblemManager).toSelf();
+    testContainer.bind(MarkerOptions).toConstantValue(PROBLEM_OPTIONS);
+    testContainer.bind(FileService).toConstantValue(<FileService>{
+        onDidFilesChange: Event.None
+    });
+
+    testContainer.bind(ProblemTree).toSelf().inSingletonScope();
+
+    rootNode = {
+        visible: false,
+        id: 'theia-' + 'problem' + '-marker-widget',
+        name: 'MarkerTree',
+        kind: 'problem',
+        children: [],
+        parent: undefined
+    };
+});
+
+beforeEach(() => {
+    rootNode = {
+        visible: false,
+        id: 'theia-' + 'problem' + '-marker-widget',
+        name: 'MarkerTree',
+        kind: 'problem',
+        children: [],
+        parent: undefined
+    };
+});
+
+after(() => {
+    disableJSDOM();
+});
+
+describe('Problem Composite Tree Node', () => {
+
+    describe('#sortMarkersInfo', () => {
+        describe('should sort markersInfo based on the highest severity', () => {
```

Comment:
minor: do you mind adding newlines after `describe` and in between `it` statements?